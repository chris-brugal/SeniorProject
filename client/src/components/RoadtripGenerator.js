import React from 'react';
import { Card, Typography, Box, TextField, CardContent, CardActions, Button } from '@mui/material';

const RoadtripGenerator = () => {
  return (
    <div>
        <Box
          display="flex"
          justifyContent="center"
          alignItems="center"
          minHeight="100vh"
          sx={{ backgroundImage: `url(${"/images/road.jpg"})`}}
        >
          <Card variant="outlined" sx={{ height: 500, width: 500 }}>
            <CardContent>
              <Typography variant="h3">Roadtrip Generator</Typography>
            </CardContent>
            <CardActions>
              <TextField id="est-trip-time" label="Estimated Trip Time" variant="outlined" />
            </CardActions>
            <CardActions>
              <TextField id="num-passengers" label="Number of passengers" variant="outlined" />
            </CardActions>
            <CardActions>
              <Button variant="contained">Submit</Button>
            </CardActions>
          </Card>
        </Box>
    </div>
    
  )
}

export default RoadtripGenerator