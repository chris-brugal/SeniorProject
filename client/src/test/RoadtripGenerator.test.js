import React from 'react';
import RoadtripGenerator from '../components/RoadtripGenerator';
import renderer from "react-test-renderer";

describe("snapshot testing", () => {

test("snapshot testing for Roadtrip Generator component", () => {
    const comp = renderer.create(<RoadtripGenerator />).toJSON()
    expect(comp).toMatchSnapshot()
})
})