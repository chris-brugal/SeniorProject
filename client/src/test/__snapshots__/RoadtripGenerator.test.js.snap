// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot testing for Roadtrip Generator component 1`] = `
<div>
  <div
    className="MuiBox-root css-rn9z1p"
  >
    <div
      className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-qyvj9t-MuiPaper-root-MuiCard-root"
    >
      <div
        className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <h3
          className="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        >
          Roadtrip Generator
        </h3>
      </div>
      <div
        className="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
      >
        <div
          className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="est-trip-time"
            id="est-trip-time-label"
          >
            Estimated Trip Time
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="est-trip-time"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-yjsfm1"
              >
                <span>
                  Estimated Trip Time
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
      >
        <div
          className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="num-passengers"
            id="num-passengers-label"
          >
            Number of passengers
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="num-passengers"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-yjsfm1"
              >
                <span>
                  Number of passengers
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Submit
        </button>
      </div>
    </div>
  </div>
</div>
`;
